README.md

```diff
@@ -139,9 +146,24 @@ And given the value to check:
 
 "FOO:123" will be validated against `CURIE(prefix.prefix)` and "a" will be validated against `in("a", "b", "c")`.
 
+### sub
+
+Usage: `sub(s/pattern/replacement/[flags], expr)`
+
+This function uses regex substitution on the contents of the target column to replace `pattern` with `replacement`. You may include optional regex flags at the end of the pattern to dictate how the pattern should match. The following flags are currently supported:
+
+* `a`: enable ASCII matching; `\w`, `\W`, `\b`, `\B`, `\d`, `\D`, `\s` and `\S` match only ASCII characters
```

Comment:
These are Python-specific flags, right? Can we handle `a` and `x` in JavaScript, for example?

Reply:
This is a subset of the [Perl regex modifiers](https://perldoc.perl.org/perlre#Modifiers). I did not include:
* `p` (prematch/match/postmatch) - "In Perl 5.20 and higher this is ignored"; also, seems to have no support in Python, although we could probably write our own support
* `d` (default) - Perl docs call this "problematic"; default behaviour in Python is Unicode matching so this seemed to not make any sense
* `l` (locale) - the Python docs advise never to use this one; use ASCII or Unicode instead
* `u` (unicode) - this is the default behaviour for python regex - I'm not sure if it is the case for JS, so that's something to double check
* `n` (nongrouping) - prevents $1, $2, etc. from being filled in; not supported in Python `re` (I don't think these get filled in in sub anyway?), but again, we could write our own support if we really wanted to

Reply:
Here are the JS flags - https://javascript.info/regexp-introduction#flags

Looks like `u` is an option here but `a` isn't, so maybe it's the opposite. What do we want the default to be? I think it makes sense to have Unicode be the default, so we always pass `u` to the JS regex, unless `a` is included in the flags. It seems like `x` isn't supported unless we use [XRegExp](http://xregexp.com/flags/)

Reply:
That sounds good.